File: packages/java/ygo74-agent-runtime/src/main/java/com/ygo74/agentruntime/domains/handlers/ResponseValidator.java
package com.ygo74.agentruntime.domains.handlers;

import com.ygo74.agentruntime.domains.contracts.ExchangeModels.StandardExchangeResponse;

public final class ResponseValidator {
    private ResponseValidator() {}

    public static void validate(StandardExchangeResponse response) {
        if (!"success".equals(response.status()) && !"error".equals(response.status())) {
            throw new IllegalArgumentException("status must be success or error");
        }
    }
}

File: packages/dotnet/Ygo74.AgentRuntime/Domains/Handlers/ResponseValidator.cs
using Ygo74.AgentRuntime.Domains.Contracts;

namespace Ygo74.AgentRuntime.Domains.Handlers;

public static class ResponseValidator
{
    public static void Validate(StandardExchangeResponse response)
    {
        if (response.Status is not ("success" or "error"))
        {
            throw new ArgumentException("status must be success or error", nameof(response));
        }

        if (response.Status == "success" && response.Output is null)
        {
            throw new ArgumentException("success response must include output", nameof(response));
        }

        if (response.Status == "error" && response.Error is null)
        {
            throw new ArgumentException("error response must include error", nameof(response));
        }
    }
}
